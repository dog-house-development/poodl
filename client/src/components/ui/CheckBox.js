import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CheckBox extends Component {
    static propTypes = {
        label: PropTypes.string,
        onChange: PropTypes.func.isRequired,
        value: PropTypes.bool,
        id: PropTypes.string.isRequired
    };

    render() {
        return (
            <div className="checkbox-wrapper">
                <label>
                    <input
                        type="checkbox"
                        id={this.props.id}
                        name={this.props.id}
                        checked={this.props.value}
                        onChange={this.props.onChange}
                    />
                    <p className="checkbox-label">{this.props.label}</p>
                </label>
            </div>
        );
    }
}

export default CheckBox;
