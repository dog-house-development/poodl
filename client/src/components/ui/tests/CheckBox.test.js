import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import CheckBox from '../CheckBox';

configure({ adapter: new Adapter() });

describe('CheckBox tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = (_props = {}) => {
        const props = _.assign(
            {},
            {
                onChange: () => {},
                label: 'Test Box',
                value: true,
                id: 'primary'
            },
            _props
        );
        wrapper = shallow(<CheckBox {..._.assign({}, props)} />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render normal checkbox correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
