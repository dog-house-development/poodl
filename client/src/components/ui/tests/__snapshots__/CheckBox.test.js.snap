// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox tests render should render normal checkbox correctly 1`] = `
<div
  className="checkbox-wrapper"
>
  <label>
    <input
      checked={true}
      id="primary"
      name="primary"
      onChange={[Function]}
      type="checkbox"
    />
    <p
      className="checkbox-label"
    >
      Test Box
    </p>
  </label>
</div>
`;
